### PR TITLE
Implement timely vote credits feature.

### PR DESCRIPTION
**Pull request by bji**
Re-submission of the timely_vote_credits feature, with a guard around credits accounting to ensure that credit counting is not altered before the feature is enabled.

### CODE REVIEW

File: programs/vote/src/vote_state/mod.rs [hunk 1]
```diff
@@ -609,41 +610,58 @@ pub fn process_new_vote_state(
...
     let mut new_vote_state_index = 0;
 
-    // Count the number of slots at and before the new root within the current vote state lockouts.  Start with 1
-    // for the new root.  The purpose of this is to know how many slots were rooted by this state update:
-    // - The new root was rooted
-    // - As were any slots that were in the current state but are not in the new state.  The only slots which
-    //   can be in this set are those oldest slots in the current vote state that are not present in the
-    //   new vote state; these have been "popped off the back" of the tower and thus represent finalized slots
-    let mut finalized_slot_count = 1_u64;
+    // Accumulate credits earned by newly rooted slots.  The behavior changes with timely_vote_credits: prior to
```
Comment (bji) [inline programs/vote/src/vote_state/mod.rs:613]: Here's the updated comment for the additional credits accounting fix.

File: programs/vote/src/vote_state/mod.rs [hunk 2]
```diff
@@ -609,41 +610,58 @@ pub fn process_new_vote_state(
...
+    // this feature, there was a bug that counted a new root slot as 1 credit even if it had never been voted on.
+    // timely_vote_credits fixes this bug by only awarding credits for slots actually voted on and finalized.
+    let timely_vote_credits = feature_set.map_or(false, |f| {
+        f.is_active(&feature_set::timely_vote_credits::id())
+    });
+    let mut earned_credits = if timely_vote_credits { 0_u64 } else { 1_u64 };
```
Comment (bji) [inline programs/vote/src/vote_state/mod.rs:619]: This line is changed from the original timely_vote_credits submission as part of the credits accounting fix.

**Comment by bji** [comment]
I have added comments to the sections that were changed from the original timely_vote_credits change.  These are the only differences with that original PR, otherwise this PR is identical to the timely_vote_credits change already accepted (I authored this new change with: `git revert -n 427b8b133203dd363a4b2a22018abd66ae3693d4` then added in the fix plus new test case).

**Comment by AshwinSekar** [comment]
Currently running this against mnb, will approve if there are no issues over the weekend

**Review by AshwinSekar — APPROVED** [review]
Confirmed no mnb divergence over the weekend with the fix.

**Pull request merged**